networking/external_dns_operator/nw-control-dns-records-public-managed-zone-gcp.adoc

```diff
@@ -0,0 +1,148 @@
+:_content-type: PROCEDURE
+[id="nw-control-dns-records-public-hosted-zone-gcp_{context}"]
+= Creating DNS records on an public managed zone for GCP by using Red Hat External DNS Operator
+
+.Procedure
+
+. Export the `kubeconfig` file:
++
+[source,terminal]
+----
+$ export KUBECONFIG=~/<path to>/kubeconfig
+----
+
+. Check the user. The user must have access to the `kube-system` namespace if you don’t have the credentials as you can fetch the credentials from the `kube-system` namespace to use the cloud provider client:
++
+[source,terminal]
+----
+$ oc whoami
+----
++
+.Example output
+[source,terminal]
+----
+system:admin
+----
+
+. Copy the value of service_account.json in gcp-credentials secret in a file encoded-gcloud.json by running the following command:
++
+[source,terminal]
+----
+$ oc get secret gcp-credentials -n kube-system --template='{{$v := index .data "service_account.json"}}{{$v}}' | base64 -d - > decoded-gcloud.json
+----
+
+. Export Google credentials by running the following command:
++
+[source,terminal]
+----
+$ export GOOGLE_CREDENTIALS=decoded-gcloud.json
+----
+
+. Activate your account by using the following command:
++
+[source,terminal]
+----
+$ gcloud auth activate-service-account  <client_email as per decoded-gcloud.json> --key-file=decoded-gcloud.json
+----
+
+. Set your project by running the following command:
++
+[source,terminal]
+----
+$ gcloud config set project <project_id as per decoded-gcloud.json>
+----
+
+. Get the routes to check the domain by running the following command:
++
+[source,terminal]
+----
+$ oc get routes --all-namespaces | grep console
+----
++
+.Example output
+[source,terminal]
+----
+openshift-console          console             console-openshift-console.apps.qe.gcp.devcluster.openshift.com                       console             https   reencrypt/Redirect     None
+openshift-console          downloads           downloads-openshift-console.apps.qe.gcp.devcluster.openshift.com                     downloads           http    edge/Redirect          None
+----
+
+. Get your zone that installer created for you. Run the following command:
++
+[source,terminal]
+----
+$ gcloud dns managed-zones list | grep qe.gcp.devcluster.openshift.com
+qe-cvs4g-private-zone qe.gcp.devcluster.openshift.com
```

Comment:
I'm not sure if we should add these `gcloud` commands in the document?
And maybe better to replace the real QE domain name with `test.gcp.example.com`?
@alebedev87  WDYT?

Reply:
This is what I tried to discuss in WIP doc (cannot find the comment anymore) but didn't get any reply.
The checks of what was done by ExternalDNS are different for AWS and Azure/GCP. In AWS we just send the user to the web UI of the provider while for Azure and GCP we provider client commands to check in details.

I'm fine with either approach: AWS one is short and easy to document, Azure/GCP is longer but gives more details. I let @xenolinux decide which approach (or mix) is better from the reader perspective.

@lihongan : Azure doc was using a QE domain too. I think it's indeed better to use some made up domain name which doesn't show any internal Red hat kitchen.

Reply:
Addressed. Replaced real QE domain name with `test.gcp.example.com`

Reply:
@alebedev87 I'm sorry. I didn't get the idea about the approach. Do we need any additional commands in the AWS document?